.changeset/loud-needles-search.md

```diff
@@ -0,0 +1,5 @@
+---
+'@shopify/hydrogen-react': patch
+---
+
+Update the TypeScript types for `<Money/>` so that the default rendered element is a `"div"`.
```

Comment:
LGTM, though I'm curious to know the motivation?

Reply:
Without the default of `= "div"` in the TS definition there (which matches the code, which defaults to `"div"` if you don't pass in an `as` prop), then TS can't know what we're rendering and it won't autocomplete props for divs like `"className"` and stuff. 

If that makes sense?